### DISCUSSION TITLE
uploading photo issues

### DISCUSSION THREAD

**Discussion by woejefe** [AlbertoV5/tosclib#14] [Q&A]
I have installed and am running tosclib in python but when i try to upload image i get this error

`File "C:\Program Files\Python310\lib\site-packages\tosclib\tosc.py", line 110, in __init__
    self.properties = self.getCreate("properties")
  File "C:\Program Files\Python310\lib\site-packages\tosclib\tosc.py", line 128, in getCreate
    s = self.node.find(target)
AttributeError: 'NoneType' object has no attribute 'find'`

the code runs all the way up to the last if statement

**Comment by AlbertoV5** [comment]
Hi, sorry it's been a year since I worked on this library. Is this running the code example for images? Sounds like it could't find the root node, do you have an example of how you are inputing the tosc file and the image? Thanks.

**Reply by woejefe** [reply]
import tosclib as tosc
from PIL import Image # type: ignore
import numpy as np
import time

from tosclib.elements import ControlType


class ImageConverter:
    def __init__(self, inputFile, outputFile, imagePath, canvasName="canvas"):
        """Stores the conversion settings"""
        self.input_path = inputFile
        self.output_path = outputFile
        self.image_path = imagePath
        self.image_size = 64
        self.pixel_size = 4
        self.canvas_name = canvasName
        self.pixels = None

    def pixelateImage(self):
        """Pixelates image and converts it into a numpy array of size (h,w,3)"""
        img = Image.open(self.image_path)
        ratio = min(img.size) / self.image_size
        xyr = int(img.size[0] / ratio), int(img.size[1] / ratio)
        imagePixelated = img.resize(xyr, resample=Image.Resampling.BILINEAR)
        self.pixels = np.asarray(imagePixelated) / 255

    def drawPixelBoxes(self):
        """Draws an array of pixels as box objects in a 'canvas' object.
        The canvas is any {self.canvas_name} named object in the top layer.
        """
        root = tosc.load(self.input_path)
        main = tosc.ElementTOSC(root[0])
        canvas = tosc.ElementTOSC(main.findChildByName(self.canvas_name))

        pxs = self.pixel_size
        for iy, ix in np.ndindex(self.pixels.shape[:2]):
            box = tosc.ElementTOSC(canvas.createChild(ControlType.BOX))
            (r, g, b) = self.pixels[iy, ix]
            box.setName(f"p{ix}{iy}")
            box.setColor((r, g, b, 1))
            box.setFrame((ix * pxs, iy * pxs, pxs, pxs))

        return tosc.write(root, self.output_path)


def main(inputFile, outputFile, imageFile, target):
    converter = ImageConverter(inputFile, outputFile, imageFile, target)
    converter.image_size = 64
    converter.pixel_size = 4
    converter.pixelateImage()
    converter.drawPixelBoxes()


if __name__ == "__main__":
    start = time.process_time()
    main(
        "C:/Users/Jefe/Desktop/Touch/test.tosc",
        "C:/Users/Jefe/Desktop/Touch/out.tosc",
        "C:/Users/Jefe/Desktop/Touch/tv.jpg",
        "canvas",
    )
    end = time.process_time()
    print("Converter", end - start)

**Reply by woejefe** [reply]
I have no idea why the code isnt staying between the ``. but that post is the full python script i am running... it is pointing to a file location on my desktop
